mne/decoding/receptive_field.py

```diff
@@ -203,13 +218,37 @@ def fit(self, X, y):
 
         # Create input features
         X, y = self._delay_and_reshape(X, y)
+
         self.estimator_.fit(X, y)
-        del X, y
         coef = get_coef(self.estimator_, 'coef_')  # (n_targets, n_features)
-        shape = [n_feats, len(self.delays_)]
+        shape = [n_feats, n_delays]
         if self._y_dim > 1:
             shape.insert(0, -1)
         self.coef_ = coef.reshape(shape)
+
+        # Inverse-transform model weights
+        if self.patterns:
+            if isinstance(self.estimator_, TimeDelayingRidge):
+                cov_ = self.estimator_.cov_ / float(n_times * n_epochs - 1)
+                y = y.reshape(-1, y.shape[-1], order='F')
+            else:
+                X = X - X.mean(0, keepdims=True)
+                cov_ = np.cov(X.T)
```

Comment:
Can you change the `TimeDelayingRidge` class to have a `patterns` option? Then this can be stored instead of being recomputed.

Reply:
could create a `_compute_patterns(model, X, y)` that would be shared between them.

Reply:
@larsoner Not sure I understand. What is being **re**computed here ? I'm only computing the pattern once either way...

Do you want the inverse pattern to be computed inside TDR if `pattern=True` ? That means there's going to be some code redundancy between receptive_field.py and TDR.py, is that OK?

Reply:
@larsoner Sorry if I'm being slow but can you elaborate on what it is you want again? 

Or are you happy with the current state of the code ?

Reply:
It looks like you've already changed the code to be fast for TDR so I think we're good